### PR TITLE
fix the routing for /users/{user_id}/tasks/{task_id} in React

### LINKED ISSUE #241: UI bug with navigation
**Describe the bug**
Bug: UI for the https://nscstrdevusw2tuecommon.z5.web.core.windows.net/users/74 shows 'The requested content does not exist.' without navigating to it. If we copy and paste the URL into another tab, we get the error. This is true for all the APIs like users, users/user-id, users/user-id/tasks, and users/user-id/tasks/task-id 

**To Reproduce**
1. Go to 'https://nscstrdevusw2tuecommon.z5.web.core.windows.net/'
2. Click on 'Users'
3. Click on 'eizon1' (or any of the users)
4. Copy the URL
5. Paste the URL onto another tab
6. See error

**Expected behavior**
UI to show the content of the users information - userid, user last name, user first name

**Screenshots**
Screenshot of the UI with manual navigation
image
Screenshot of the UI after copying and pasting the URL
image

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser: Firefox
 - Version: 86.0 (64-bit)

**Smartphone (please complete the following information):**
- NA

**Additional context**
This bug applies to all the APIs

### LINKED ISSUE #299: Fix the routing for /users/{user_id} in React
fix the routing for /users/{user_id} in React 

## Problem:
When client requests homepage and further navigates to https://nscstrdevusw2tuecommon.z5.web.core.windows.net/users/43 app behaves as expected. However when this URL is copied and opened in another browser it sends 404 error.

## Why: 
The server can't recognized the request. Our plan was not to implement it on the server but use the client side. 

## Solution 
Implementing the HashRouter so our url will look like this: https://nscstrdevusw2tuecommon.z5.web.core.windows.net/#/users/43 
This URL works by sending the request to server of the home page only, every single time. And the routing is done on the client side. 
This way the URL can be copied and shared with no 404 error. 

| Date | Activity| Time |
| :---:        |     :---:      |     :---: |
| 3/07  | Researched HashRouter    |  2 hr  |
| 3/08  | Review the code, receiving error 500 |  1 hr  |
| 3/09   |  Created task on Github  |  5 min  |
|  3/10  | Actions are breaking functions, can't test, created plan for cooperation between other groups    | 1 hrs    |
|  3/11  | Functions were working so tested HashRouter locally | 1 hrs    |
| 3/12 | Debating with group about the results read more about client-side routing | 1 hrs |
| 3/13 | functions again broken, can't test, emailed asking for manual function deployment and checking if this bug has an owner  | 0.5 hrs |
| 3/15 | testing locally Juno's HashRouter implementation as well as in Firefox browser receiving 500 error for functions | 0.5 hrs |

### LINKED ISSUE #309: fix the routing for /users/{user_id}/tasks/{task_id} in React
this issue fixes the routing of the React app hosted by Azure by using HashRouter. 

## What:
Fixed the bug described here. 
https://github.com/North-Seattle-College/ad440-winter2021-tuesday-repo/issues/241

## How to use:

Steps to check the behavior:

    Go to 'https://nscstrdevusw2tuecommon.z5.web.core.windows.net/'
    Click on 'Users'
    Click on 'eizon1' (or any of the users)
    Click refresh
    See the page refreshed without seeing any error

## Local testing:

 N/A for this behavior because localhost:// does not reproduce the same routing error

## Remote testing:
  Check How to use

## Why:

We caught this error because UI tests with screenshots were all failing and seeing 404 error. This means whenever an user refreshes or sends a nested link of the website to another user our website will give them error, causing possibly the worst user experience.  

## Timing:
Date | Activity | Time
-- | -- | --
2/08 | Researched and read about HashRouter | 1 hours
2/08 | Modified the code locally for HashRouter | 0.5 hours
2/14 | Created task on Github | 2 min
2/14 | Read about difference between server side routing and client side routing | 1 hours
2/15 | Caught another bug within the web app and resolved it with the team | 1 hours
2/16 | Pushed the changes and tested remotely | 0.5 hours

### PR DESCRIPTION
**Pull request by OyunominJuno**

### CODE REVIEW

File: ui/src/App.js [hunk 1]
```diff
@@ -4,6 +4,8 @@ import {
   Switch,
   Redirect,
   BrowserRouter as Router,
```
Comment (asiagromasia) [inline ui/src/App.js]: Suggestion: to remove unneeded import

**Review by asiagromasia — COMMENTED** [review]
At this moment there is a bug in src/components/pages/TaskListScreen.js  "onDeleteHandler". 
Suggestion: remove brackets from PR name

File: ui/src/App.js (hunk 1 above)
Comment (OyunominJuno) [reply ui/src/App.js]: Done this, thank you

**Comment by jlsnow301** [comment]
Could you write a description for one of these? I'm confused as to what this is fixing

**Review by asiagromasia — APPROVED** [review]
let's test it

**Comment by asiagromasia** [comment]
the bug was fixed so we can use this route now

**Pull request merged**